### ISSUE TITLE
Geometry subtype not inferred for function returns

### ISSUE THREAD

**Issue by apjoseph** [graphile/postgis#41]
With
```
    "postgraphile": "4.12.2",
    "@graphile/postgis": "0.1.0",
```

The following function will return a geometry interface as a return type instead of a point geometry. This is particularly frustrating for computed columns on tables (for example generating a centroid for a polygon geometry).

```pgsql
CREATE FUNCTION myschema.mytable_pos(mytable myschema.mytable) RETURNS geometry(Point,4326) AS $$
SELECT st_pointonsurface(geom)
$$ LANGUAGE sql STABLE
````

**Comment by benjie** [comment]
Annoyingly this is a limitation of PostgreSQL. It may look like it returns that type in your definition but if you look in a pg_dump of the database you’ll notice the modifier is dropped. PostgreSQL does not store this.

**Comment by apjoseph** [comment]
@benjie ah, ok so postgres is not even storing the return type modifiers -that is bizarre behavior on the part of postgres as type modifiers are pretty important!

Maybe for these types of cases an `@forceType `  tag would be beneficial so that the modifiers could be manually specified.

**Comment by benjie** [comment]
Yeah, implementing it would be tricky since the modifier is a 4 byte binary value under the hood. Effectively you’d have to encode the human readable version back to the byte representation so that PostGraphile could convert it back again; and it would only be useful for postgis since every type processes it’s modifiers differently. If you’re interested in taking this on; here’s a starting point:

https://github.com/graphile/postgis/blob/master/src/utils.ts#L4
